### PR TITLE
Add Python language support and migrate Rust to macro pattern

### PR DESCRIPTION
**Pull request by loganmoon**
## Summary
- Migrate Rust extractor to use `define_language_extractor!` macro for consistency with TypeScript/JavaScript
- Rename `handlers/` to `handler_impls/` directory to match established convention
- Implement full Python language support with function, class, and method entity extraction

## Python Features
- Function extraction with parameters, return types, and type hints
- Class extraction with base classes and inheritance
- Method extraction including decorated methods (`@staticmethod`, `@classmethod`, `@property`)
- Docstring extraction
- Decorator extraction
- Async function detection
- Variadic parameters (`*args`, `**kwargs`)
- Self/cls parameter filtering for methods
- Support for `.py` and `.pyi` stub files

## Test plan
- [x] All 119 Rust language tests pass
- [x] All 19 new Python integration tests pass
- [x] All 8 JavaScript integration tests pass
- [x] All 11 TypeScript integration tests pass
- [x] All 29 indexer tests pass (updated to expect Python support)
- [x] Clippy passes with no warnings
- [x] cargo fmt passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by loganmoon — COMMENTED** [review]
## Code Review Summary

Overall this is a well-structured PR that adds Python language support following the established macro-based pattern. The code demonstrates good separation of concerns, proper error handling, and comprehensive test coverage.

**Positive observations:**
- Clean macro design reduces boilerplate effectively
- Thorough test coverage (19 integration tests + 3 unit tests)
- Smart handling of self/cls parameter filtering
- Good error handling with Result types throughout
- Follows established patterns from JavaScript/TypeScript handlers

File: crates/languages/src/common/python_common.rs [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+                            node_to_text(n, source).ok()
+                        } else {
+                            None
+                        }
+                    })
+                    .unwrap_or_default();
+                let type_hint = child
+                    .child_by_field_name("type")
+                    .and_then(|n| node_to_text(n, source).ok());
+                parameters.push((name, type_hint));
```
Comment (loganmoon) [inline crates/languages/src/common/python_common.rs]: Using `.unwrap_or_default()` here converts extraction failures to empty strings silently. This could lead to parameters with empty names being added to the list.

Consider either skipping parameters that fail to extract (using `continue`) or propagating the error. The same pattern appears in `typed_parameter`, `default_parameter`, and `typed_default_parameter` cases below.

**Comment by loganmoon** [comment]
## Code Review Summary

Good implementation overall. The macro pattern migration is clean, and Python support is comprehensive. The following issues are worth addressing:

### Issues to Address

1. **CLAUDE.md is outdated** - states Python has "no actual parsing implementation" but this PR implements full support. This should be updated to reflect Python is now fully implemented.

2. **Silent error handling for file paths** - Using `unwrap_or_default()` for path conversion silently produces empty strings for non-UTF8 paths, which could lead to incorrect entity IDs. Consider propagating errors or logging.

3. **Missing edge case tests** - No tests for empty source files, malformed syntax, or Python 3.8+ positional-only/keyword-only parameters (`def f(a, /, b)` or `def f(*, a)`). These parameters would be silently dropped.

4. **Unused function** - `is_method_context` in `python_common.rs` is defined but never called.

### Minor Notes

- Decorator detection uses magic strings; consider constants like Rust's `constants.rs`
- Python handlers duplicate entity building logic that could be extracted into a shared helper

No security issues found. Performance is acceptable.

File: crates/languages/src/python/handler_impls/function_handlers.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    // Build signature
+    let signature = FunctionSignature {
+        parameters,
+        return_type,
+        generics: Vec::new(),
+        is_async,
+    };
+
+    // Generate entity_id
+    let file_path_str = file_path.to_str().unwrap_or_default();
```
Comment (loganmoon) [inline crates/languages/src/python/handler_impls/function_handlers.rs]: Using `unwrap_or_default()` here silently produces an empty string for non-UTF8 paths, which would create incorrect entity IDs. Consider propagating the error (like Rust handlers do at `rust/handler_impls/common.rs:354`) or at least logging a warning.

File: crates/languages/src/common/python_common.rs [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+                let name = child
+                    .named_child(0)
+                    .and_then(|n| node_to_text(n, source).ok())
+                    .map(|n| format!("**{n}"))
+                    .unwrap_or_else(|| "**kwargs".to_string());
+                parameters.push((name, None));
+            }
+            _ => {}
+        }
+    }
```
Comment (loganmoon) [inline crates/languages/src/common/python_common.rs]: The catch-all `_ => {}` silently ignores Python 3.8+ positional-only (`def f(a, /, b)`) and keyword-only (`def f(*, a)`) parameters. Consider adding explicit handling or at least logging when unknown parameter types are encountered.
